site/src/components/Workspace/Workspace.tsx

```diff
@@ -131,6 +132,40 @@ export const Workspace: FC<React.PropsWithChildren<WorkspaceProps>> = ({
   if (template !== undefined) {
     transitionStats = ActiveTransition(template, workspace)
   }
+
+  const [showAlertPendingInQueue, setShowAlertPendingInQueue] = useState(false)
+  const now = dayjs()
+  useEffect(() => {
+    if (
+      workspace.latest_build.status === "pending" &&
+      workspace.latest_build.job.queue_size > 0 &&
+      dayjs(workspace.latest_build.created_at).isBefore(
+        now.subtract(5, "seconds"),
+      )
+    ) {
+      setShowAlertPendingInQueue(true)
+      return
+    }
+
+    if (
+      workspace.latest_build.status === "pending" &&
+      workspace.latest_build.job.queue_size > 0
+    ) {
+      const timer = setTimeout(() => {
+        if (
+          workspace.latest_build.status !== "pending" ||
+          workspace.latest_build.job.queue_size === 0
+        ) {
+          return
+        }
+        setShowAlertPendingInQueue(true)
+      }, 5000)
```

Comment:
How about using the difference here (now.subtract 5 sec) so that loading a new workspace doesn't reset the timer to 5 seconds?

Reply:
So let's say you:

1. Start a workspace build
2. Wait 3 seconds
3. Reload the workspace page
4. Build still pending, we start the timer with hard-coded 5 second wait
5. Queue banner is shown (time since build started 8 seconds vs 5 seconds)

Does this make sense?

Reply:
Make sense now, thanks.

I used the _diff_, feel free to review that part.